### ISSUE TITLE
SELinux support

### ISSUE THREAD

**Issue by matiasinsaurralde** [TykTechnologies/tyk#803]
**Do you want to request a *feature* or report a *bug*?**
Enhancement

**What is the current behavior?**
Tyk doesn't provide SELinux policies.

**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
All.

It should be possible to provide SELinux policies [as part of our RPM packages](https://access.redhat.com/discussions/1171373), a process like [this](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/4/html/SELinux_Guide/selg-section-0128.html) can be used to write the policies.

**Comment by pss-john** [comment]
@matiasinsaurralde and @weetat 

What is the directory structure of the RPM?  I can not find it, I'm testing out different technologies, I've already written one for traefik, and since traefik didn't meet my requirements I'm testing out tyk

My deployment strategy is forward all port 80 and 443 to 8080 and 8443 through firewalld so the process isn't ran as root.

Example from what I auto-generated with sepolicy  and audit2allow with some hand tweaking, for  Traefik (although audit2allow just popped up with a rule I missed):

```
/usr/bin/traefik		--	gen_context(system_u:object_r:traefik_exec_t,s0)

/var/log/traefik(/.*)?	    gen_context(system_u:object_r:traefik_log_t,s0)

/etc/traefik(/.*)?          gen_context(system_u:object_r:traefik_conf_t,s0)
```

```
policy_module(traefik, 1.0.0)

require {
    type unconfined_t;
    class dir search;
}

########################################
#
# Declarations
#

type traefik_t;
type traefik_exec_t;
init_daemon_domain(traefik_t, traefik_exec_t)

permissive traefik_t;

type traefik_log_t;
logging_log_file(traefik_log_t)

########################################
#
# traefik local policy
#
allow traefik_t self:fifo_file rw_fifo_file_perms;
allow traefik_t self:unix_stream_socket create_stream_socket_perms;


manage_dirs_pattern(traefik_t, traefik_log_t, traefik_log_t)
manage_files_pattern(traefik_t, traefik_log_t, traefik_log_t)
manage_lnk_files_pattern(traefik_t, traefik_log_t, traefik_log_t)
logging_log_filetrans(traefik_t, traefik_log_t, { dir file lnk_file })

domain_use_interactive_fds(traefik_t)


type traefik_conf_t;
traefik_manage_conf(unconfined_t, traefik_conf_t, traefik_conf_t)
manage_dirs_pattern(traefik_t, traefik_conf_t, traefik_conf_t)
manage_files_pattern(traefik_t, traefik_conf_t, traefik_conf_t)
miscfiles_read_localization(traefik_t)

allow init_t traefik_conf_t:dir search;
init_nnp_daemon_domain(traefik_t)
```

**Comment by alephnull** [comment]
@pss-john `rpm -qpl some.rpm` gives a list of the paths in an RPM. `rpm -ql` does the same for a package that is installed.

The binary is `/opt/tyk-gateway/tyk`, btw.

**Comment by pss-john** [comment]
@alephnull I don't even know where the RPM is located, when I read the documentation the RHEL instructions pointed to the Ansible instructions. So I just built it myself... oddly enough using an Ansible role I wrote.

Also how does Tyk typically connect to port 80/443? Does it use CAP_NET_BIND_SERVICE or similar or does it use a privileged process that transmits to unprivileged worker process like Nginx?
